### DISCUSSION TITLE
Improvements for Django support

### DISCUSSION THREAD

**Discussion by patrick91** [strawberry-graphql/strawberry#630] [Ideas]
Hi folks, I was working on a project for my parents and I though we can improve our Django support (excluding the model conversion part for now).

Ideas:

- [ ] Built in support for JWT
- [ ] Websockets support

What do you think? Anything else we should add?

**Comment by ulgens** [comment]
I was just to write about model serializer :slightly_smiling_face:

**Reply by patrick91** [reply]
That's something I think we should add, in future 😊 I'd love to have something that allows to create CRUD mutations from django models, but we need to iron out the basics first (and Pydantic support) :)

**Reply by hansek** [reply]
@patrick91 is there any implementation of model serialization you know about?

**Comment by ulgens** [comment]
From existing issues:

https://github.com/strawberry-graphql/strawberry/issues/144
https://github.com/strawberry-graphql/strawberry/issues/162

**Comment by jkimbo** [comment]
> Built in support for JWT

Why built in support for JWT? Is that a common way to handle authentication in Django? I've only used the built in session management.

> Websockets support

+1 on this. Is Channels still the best way to add websocket support to Django?

**Reply by ulgens** [reply]
> Why built in support for JWT? Is that a common way to handle authentication in Django? I've only used the built in session management.

@jkimbo For the last 2-3 years, it was the only auth method we used in our projects in multiple companies. We had to use a third-party package with graphene for that. I'd be very happy to have native support in Strawberry.

**Reply by patrick91** [reply]
@jkimbo for websockets you can use ASGI :)

For JWT, I'm interested in your opinions, it could also be a separate library if it makes sense. But providing something would be pretty useful for others I think. Ideally we could share most of the core with Flask as well or make it pluggable (ie with different dbs).

**Reply by ulgens** [reply]
From my experience, on JWT, if someone / some team are passionate enough to graphql with Python in a project, they don't go with classical auth token 😄

**Reply by jkimbo** [reply]
> From my experience, on JWT, if someone / some team are passionate enough to graphql with Python in a project, they don't go with classical auth token 😄

I disagree with that. I’ve never found JWTs particularly compelling for simple authentication. It also seems to have constant issues so I’ve generally avoided it. 

Can’t deny that it is popular though so I’m all for building support for it into Strawberry.

**Comment by matclayton** [comment]
Even things like adding management commands for exporting the schema would help

```py
class Command(BaseCommand):
    help = "Export GraphQL Schema"

    def handle(self, *args, **options):
        print(print_schema(schema))
```

for example, is quick and easy boilerplate, but goes a long way. We failed to get the standard command to discover the schema and export it when run under django, but this works nicely.

**Reply by jkimbo** [reply]
@la4de this is a really good point. Are you planning on adding it to the Django integration?
